pkg/config/metadata.go

```diff
@@ -112,6 +119,14 @@ func SetMetadataDefaults(jsonWithDefaults []byte) {
 		newKey := mdValuesCfgPrefix + k
 		// ex: "metadata.values.ami-id": "ami-0a887e401f7654935"
 		mdValuesDefaults[newKey] = v
+
+		// if mdvalue is a nested struct, then re-unmarshal json data to correct type
+		if nestedStruct, ok := mdNestedValues[newKey]; ok {
+			updatedVal, err := unmarshalToNestedStruct(v, nestedStruct)
+			if err == nil {
```

Comment:
what's the behavior of an error here? The metadata for elastic inference just ends up missing but otherwise we're good?

Reply:
correct. elastic inference is still accessible, but all fields will be empty.